pkg/ss/ss.go

```diff
@@ -19,7 +19,7 @@ type ConsumerFactory interface {
 }
 
 type Router interface {
-	OnMsg(m Msg) error
+	OnMsg(m Msg) (<-chan error, error)
```

Comment:
`OnMsg(m Msg) (<-chan (interface{}, error))`
Even if there is an error, handle it in async fashion. The perf overhead is justified assuming error scenarios are rare